clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1817,8 +1831,16 @@ class SyclKernelIntHeaderCreator : public SyclKernelFieldHandler {
     // sampler __init method has only one argument
     const ParmVarDecl *SamplerArg = InitMethod->getParamDecl(0);
     assert(SamplerArg && "sampler __init method must have sampler parameter");
+    uint64_t Size;
```

Comment:
Looking at the comment on 1831, I think we can skip the changes here.  It already only has 1 argument, so there is nothing to change.  Perhaps an assert somewhere that InitMethod->param_size()  == 1?